### ISSUE TITLE
WarnServ should reply to PM's as well

### ISSUE THREAD

**Issue by shymega** [vktec-irc/warnserv#3]
WarnServ (on the refactor branch, FYI) should be able to respond and
act upon commands sent via PM.

At the moment, we add warnings in-channel, which leads to tension, and
spamming in some severe cases.

ChanServ can be interacted with over PM, why not WarnServ? :-)

**Comment by vktec** [comment]
Yes, this has been a to-do for some time, but I've not gotten around to it yet.

**Comment by shymega** [comment]
OK, sounds good.

I'm trying to get it working on the refactor branch, but no luck yet.

I daresay you saw me testing it in #bots as well :)

If you come up w/ anything, feel free to commit to refactor branch.

**Comment by vktec** [comment]
You may find this piece of code useful:
```scheme
(define (message-reply-to msg)
  "Return the place to send a reply to"
  (if (string-ci= (irc:message-receiver msg)
                  config:irc:nick)
      ;; PM
      (irc:message-sender msg)
      ;; Channel message
      (irc:message-receiver msg)))
```

Use it like this (where `msg` is an `irc:message` record):
```scheme
(irc:say conn "My message" (message-reply-to msg)
```

**Comment by shymega** [comment]
OK, I'll bear that code in mind.

Not quite sure how to implement that on the (bot:cmd) procedure
though, which in turn uses irc:add-message-handler!

Think it's something to do with the 'sender' keyword.

**Comment by shymega** [comment]
Can confirm, this does now work in WarnServ's `rewrite` branch!
